packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1228,10 +1228,22 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
   /// render object indicated that it uses the size of this render object in
   /// computing its layout information, this function will also mark the parent
   /// as needing layout.
-  void markNeedsLayout() {
+  ///
+  /// The `parentDefinitelyNeedsLayout` argument can be set to true to indicate
+  /// that the parent must be notified regardless. This is used, e.g., by
+  /// [RenderBox.markNeedsLayout] to indicate that the parent has used baseline
+  /// or intrinsic dimension information from this object, so when this object
+  /// needs layout, the parent must also be dirtied.
+  void markNeedsLayout({ bool parentDefinitelyNeedsLayout: false }) {
```

Comment:
We should try to avoid adding optional parameters to these functions.  The functions are called a large number of times and having optional parameters make each call site slower.

Instead, I'd recommend just having RenderBox call parent?.markNeedsLayout() after calling super.markNeedsLayout().  It's harmless to call markNeedsLayout multiple times and that avoids a bunch of complexity.